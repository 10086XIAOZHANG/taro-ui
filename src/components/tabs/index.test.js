import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtTabs from '../../../.temp/components/tabs/index'

describe('AtTabs Snap', () => {
  const tabList = [{ title: '标签页1' }, { title: '标签页2' }, { title: '标签页3' }]

  it('render initial AtTabs', () => {
    const componet = renderToString(<AtTabs isTest />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props customStyle', () => {
    const componet = renderToString(<AtTabs isTest customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props className', () => {
    const componet = renderToString(<AtTabs isTest className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props tabList', () => {
    const componet = renderToString(<AtTabs isTest tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props current', () => {
    const componet = renderToString(<AtTabs isTest current={2} tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props swipeable', () => {
    const componet = renderToString(<AtTabs isTest swipeable tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props scroll', () => {
    const componet = renderToString(<AtTabs isTest scroll tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props tabDirection', () => {
    const componet = renderToString(<AtTabs isTest scroll tabDirection='vertical' tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props animated', () => {
    const componet = renderToString(<AtTabs isTest animated={false} tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtTabs -- props height', () => {
    const componet = renderToString(<AtTabs isTest height='300px' tabDirection='vertical' tabList={tabList} />)
    expect(componet).toMatchSnapshot()
  })
})
