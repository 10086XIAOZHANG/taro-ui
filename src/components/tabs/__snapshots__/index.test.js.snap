// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabs Snap render AtTabs -- props className 1`] = `"<div class=\\"at-tabs test\\"><div class=\\"at-tabs__header\\"></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props current 1`] = `"<div class=\\"at-tabs\\"><div class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item\\">标签页1</div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2</div><div id=\\"tab2\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页3</div></div><div style=\\"transform: translate3d(-200%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-tabs\\"><div class=\\"at-tabs__header\\"></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props scroll 1`] = `"<div class=\\"at-tabs\\"><div scrollX scrollWithAnimation scrollLeft=\\"0\\" scrollIntoView=\\"\\" class=\\"taro-scroll taro-scroll-view__scroll-x at-tabs__header at-tabs__header--scroll\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1</div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2</div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3</div></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props swipeable 1`] = `"<div class=\\"at-tabs\\"><div class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1</div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2</div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3</div></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render AtTabs -- props tabList 1`] = `"<div class=\\"at-tabs\\"><div class=\\"at-tabs__header\\"><div id=\\"tab0\\" class=\\"at-tabs__item at-tabs__item--active\\">标签页1</div><div id=\\"tab1\\" class=\\"at-tabs__item\\">标签页2</div><div id=\\"tab2\\" class=\\"at-tabs__item\\">标签页3</div></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;

exports[`AtTabs Snap render initial AtTabs 1`] = `"<div class=\\"at-tabs\\"><div class=\\"at-tabs__header\\"></div><div style=\\"transform: translate3d(-0%, 0px, 0px);\\" class=\\"at-tabs__body\\"></div></div>"`;
